Fix macOS setpup docs

rtitmuss opened the pull request.

dxmh commented on docs/docs/development/setup.md:
```diff
@@ -343,11 +343,12 @@ The install command is:
-:::warning Security Controls Workaround
+Next, we'll configure Zephyr™ with some extra environment variables needed to find the cross compiler by running the following commands:
 
-Please be sure to read the [additional setup instructions](https://docs.zephyrproject.org/2.5.0/getting_started/installation_mac.html#mac-gatekeeper) needed to address security controls found in macOS 10.15 Catalina and newer
```
I think it would be good if another macOS user could chip in regarding the removal of the security notice in https://github.com/rtitmuss/zmk/commit/3c098d065ea9c0067df2854ba2a4541cefd7330d, but it does seem to make sense as the linked page states the following:

> Note that this section does not apply to executables installed with Homebrew, since those are automatically un-quarantined by brew itself. This is however relevant for most 3rd Party Toolchains.

dxmh replied:
Ah, so it seems this security workaround is not necessary _if_ the GNU ARM Embedded is installed via the `brew` command, but the warning _is_ necessary if users install “manually” via the steps on the page linked in the previous paragraph.

rtitmuss replied:
Good catch. I have updated the PR and also removed the page linked. This is not required when following these instructions. Note also in the 'Prerequisites' section it states 'Homebrew is required to install the system dependencies'. If you choose to complete the setup without brew, then this security warning would apply earlier in the process too.

dxmh commented:
Just wanted to leave a reference here to https://github.com/zmkfirmware/zmk/pull/917 which has some similar changes to https://github.com/rtitmuss/zmk/commit/f50976f865863a6cda5f50af7dc63b7820a0b3bf.

rtitmuss commented:
> Just wanted to leave a reference here to #917 which has some similar changes to [rtitmuss@f50976f](https://github.com/rtitmuss/zmk/commit/f50976f865863a6cda5f50af7dc63b7820a0b3bf).

I added the similar changes as suggestions to #917 and removed them from this commit.

dxmh reviewed (approved):
Thanks. Given the above discussion (https://github.com/zmkfirmware/zmk/pull/932#pullrequestreview-747430665), these changes seem sensible to me. 👍
